### ISSUE TITLE
Python module installed into the bin/ directory instead of the Python module directory

### ISSUE THREAD

**Issue by pabs3** [microsoft/SPTAG#203]
When building with cmake, the Python module installed into the `bin/` directory instead of the default Python module directory. I'm not sure, but I think the right way to resolve this is to use the [Python3_add_library](https://cmake.org/cmake/help/latest/command/add_library.html#command:add_library) command from the [FindPython3](https://cmake.org/cmake/help/latest/module/FindPython3.html) cmake module. [Searching GitHub](https://github.com/search?q=Python3_add_library+NOT+my_module+NOT+ARGN&type=code) I found some examples of how to use the `Python3_add_library` function.

**Comment by pabs3** [comment]
Looks like there is also the [FindPython](https://cmake.org/cmake/help/latest/module/FindPython.html) cmake module, which abstracts away the Python version, including adding a `Python_add_library` function. I suggest dropping the custom Python version handling and using that instead. I note that the CUDA detection interacts with the Python detection, so that will need adjusting too.

**Comment by g1soori** [comment]
@pabs3 were you able to build the python SPTAG module on linux ? I was able to build it successfully but python module is not there inside the Release folder. Any help would appreciate.

**Comment by pabs3** [comment]
Yes. I have packaged SPTAG for Debian (included in bookworm), including
the Python module. The patches in my fork of SPTAG are needed for this
and other issues and are distributed with the Debian package of SPTAG.

This issue is fixed by the last commit in this branch:

https://github.com/microsoft/SPTAG/compare/main...pabs3:SPTAG:install-to-python-modules-dir

Unfortunately contributing to SPTAG requires signing a CLA,
otherwise I would contribute this patch for inclusion in SPTAG.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/

**Comment by g1soori** [comment]
Great. It worked. Thanks a lot @pabs3

**Comment by g1soori** [comment]
@pabs3 by any chance did you try install it on macos as well ? Atleast the python module ?

**Comment by pabs3** [comment]
Nope, I only tried on Debian. That said, there is nothing
Linux-specific in the patch, so it should work on macOS too.

-- 
bye,
pabs

https://bonedaddy.net/pabs3/
